src/qibolab/platforms/abstract.py

```diff
@@ -282,11 +282,29 @@ def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None):
     def __call__(self, sequence, nshots=1024, relaxation_time=None):
         return self.execute_pulse_sequence(sequence, nshots, relaxation_time)
 
-    def sweep(self, sequence, *sweepers, nshots=1024, relaxation_time=None, average=True):
+    def sweep(self, sequence, *sweepers, nshots=1024, average=True, relaxation_time=None):
         """Executes a pulse sequence for different values of sweeped parameters.
-
         Useful for performing chip characterization.
 
+        Example:
+            .. code-block:: python
+
+                import numpy as np
+                from qibolab.platform import Platform
+                from qibolab.sweeper import Sweeper
+                from qibolab.pulses import PulseSequence
+
+
+                platform = Platform("dummy")
+                sequence = PulseSequence()
+                parameter = "frequency"
+                pulse = platform.create_qubit_readout_pulse(qubit=0, start=0)
+                sequence.add(pulse)
+                parameter_range = np.random.randint(10, size=10)
+                sweeper = Sweeper(parameter, parameter_range, [pulse])
+                platform.sweep(sequence, sweeper)
```

Comment:
Just to let you know:
- in principle, to make a code block in rest, you just need: indentation and double colon the line before, i.e. `Example::` https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks (and by default it should be Python code, unless configure differently)
- if you want to make a doctest (that is incredibly useful) you have just to prepend `>>>` to your input lines https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#doctest-blocks (no need for double colon or indentation)

Reply:
Thanks. Probably the best solution here is to use the `.. testcode::` syntax, this is usually what we use in Qibo.